ui/dashboards/src/context/DashboardProvider/panel-slice.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PanelDefinition } from '@perses-dev/core';
+import { StateCreator } from 'zustand';
+import { Middleware } from './common';
+
+/**
+ * Slice with the state of Panels, along with any actions that modify only the Panels state.
+ */
+export interface PanelSlice {
+  panels: Record<string, PanelDefinition>;
+  previousPanels: Record<string, PanelDefinition>;
```

Comment:
We don't have a type for panel keys / IDs yet, I wonder if we'd want this type

Reply:
Yeah, good idea, we probably want a type alias.